### ISSUE TITLE
[Azure] Add resource_group and subscription_id labels to Prometheus metrics

### ISSUE THREAD

**Issue by boxcee** [vmware-tanzu/velero#1952]
**Describe the problem/challenge you have**
With #1895, cross subscription backups/restore were enabled. In order to monitor backups cross subscription and resource group it is required to add some labels to the metrics.

**Describe the solution you'd like**
Add more labels to prometheus metrics. E.g. resource_group and subscription_id.

**Anything else you would like to add:**
I am happy to add them by myself. However, I noticed that you are in the progress of separating velero core and provider relevant code. I would need some advice on how to start then (where to put the code).

**Comment by skriss** [comment]
@boxcee yes, we're going to be moving the cloud-provider code into separate repos. That work is in progress as we speak - we should soon have https://github.com/vmware-tanzu/velero-plugin-for-microsoft-azure` repo that contains most of the code that's currently in `pkg/cloudprovider/azure`. If you can hold off on implementing this until we do that migration, that would probably work out best for all of us. Thanks!

**Comment by skriss** [comment]
@boxcee we've finished moving the Azure code into its own repo, so it should be fine to proceed with this issue now if you're still interested.

**Comment by boxcee** [comment]
Thanks for letting me know! Will add.
